### DISCUSSION TITLE
Slow in comparing (folder/directorie)s and their files.

### DISCUSSION THREAD

**Discussion by antdude** [WinMerge/winmerge#1314] [Q&A]
Is it normal for WinMerge to be very slow when comparing many files and (folder/directorie)s. Files can huge (GB) too. These are on slow HDDs. However, FreeFileSync is very fast. Basically, I just want to compare and synchronize my datas between HDDs. This is in my 64-bit W10 Pro PC.

Thank you for reading and hopefully answering soon. :)

**Comment by Gitoffthelawn** [comment]
Does FreeFileSync perform a byte-by-byte comparison of the content of each file?  IIRC, it compares the size and timestamps, but not the content.  You might want to verify that, because I'm going by memory.  I also don't recall, if FreeFileSync finds a different timestamp for a file, if it then performs a byte-by-byte comparison of the file's contents.  It will be helpful to learn the answer to that detail.

If FreeFileSync is *not* performing byte-by-byte comparisons, and WinMerge is configured to do this, then WinMerge will artificially appear to be slower.  To rectify this, you can modify the compare method within WinMerge's *Options > Compare > Folder* UI.

If, after comparing apples to apples, WinMerge is slower than FreeFileSync, then (and only then) would it confirm that comparison speed improvements to WinMerge could be made.  If this is demonstrated to be the case, please post details so the potential issue can be further investigated and rectified.  Thanks!

**Comment by antdude** [comment]
I forgot to mention that I used "Quick Contents". That should be fast enough, right?

**Comment by sdottaka** [comment]
Could you try the 'Binary Contents' method?
The 'Full Contents' method and 'Quick Contents' method do multi-threaded comparisons.
However, on hard disks, these methods can be slower. Binary Contents' method does single-threaded comparisons and maybe faster.

The number of threads can be changed in the options dialog as follows

![image](https://user-images.githubusercontent.com/98126/165887066-459dade8-94fd-429c-8b34-adac241d5e74.png)

**Comment by antdude** [comment]
I'm using the default settings for compare method. Each one shows -1. I have a six core Intel CPU. I tried binary content method. It's still slow when scanning my huge media collection (videos, audio, images, etc. = About 245 GB with 5,622 files and 847 (folder/directorie)s. 

However, I did find using compare with "Modified Date and Size" method fast like FreeSync. Obviously, this doesn't check the checksums. I noticed FreeFileSync's compare is using this method too in its option. That explains the quickness. Now, I found the answer. Thanks you for investigating and answering. :)
